lib/blocs/calendar_cubit.dart

```diff
@@ -122,88 +132,153 @@ class CalendarCubit extends Cubit<CalendarState> {
 
   /// The offset to be used for the next paginated request.
   int _nextOffset = 0;
+  int _nextPastOffset = 0;
+
+  /// The time rounded down to the month used to split the events in "past" and
+  /// "future". Also used in the query. This is a final to avoid race conditions
+  final DateTime _splitTime =
```

Comment:
Shouldn't this be set in load() (so that force-reloading does use a new split time)?

Reply:
I could indeed update it, there, but initializing it here allows us to remove the `?`. Updating it doesnt really matter as it is rounded to the month.